packages/dds/tree/src/simple-tree/api/schemaFactoryRecursive.ts

```diff
@@ -47,13 +48,37 @@ export function createFieldSchemaUnsafe<
  * }
  * ```
  * @remarks
- * The type of a recursive schema can be passed to this, and a compile error will be produced for some of the cases of malformed schema.
+ * In this context recursive schema are defined as all {@link FieldSchema} and {@link TreeNodeSchema} schema which are part of a cycle such that walking down through each {@link TreeNodeSchemaCore.childTypes} the given starting schema can be reached again.
+ * Schema referencing the recursive schema and schema they reference that are not part of a cycle are not considered recursive.
+ *
+ * TypeScript puts a lot of limitations on the typing of recursive schema.
+ * To help avoid running into these limitations and thus getting schema that do not type check (or only type checks sometimes!),
+ * {@link SchemaFactory} provides APIs (postfixed with `recursive`) for writing recursive schema.
+ * These APIs when combined with the patterns documented below should ensure that the schema provide robust type checking.
+ * These special patterns (other than {@link LazyItem} forward references which are not recursion specific)
+ * are not required for correct runtime behavior: they exist entirely to mitigate TypeScript type checking limitations and bugs.
+ * Ideally TypeScript's type checker would be able to handle all of these cases and more, removing the need for recursive type specific guidance, rules and APIs.
+ * Currently however there are open issues preventing this:
+ * [1](https://github.com/microsoft/TypeScript/issues/59550),
+ * [2](https://github.com/microsoft/TypeScript/issues/55832),
+ * [3](https://github.com/microsoft/TypeScript/issues/55758).
```

Comment:
These should ideally be `@link` tags rather than markdown links.
```suggestion
 * {@link TODO https://github.com/microsoft/TypeScript/issues/59550 | 1},
 * {@link TODO https://github.com/microsoft/TypeScript/issues/55832 | 2},
 * {@link TODO https://github.com/microsoft/TypeScript/issues/55758 | 3}.
```

Reply:
What does the "TODO" here do?

Reply:
With that I get `tsdoc-reference-missing-dot: Expecting a period before the next component of a declaration referenceeslinttsdoc/syntax` and they don't work in intelisense.

Reply:
I updated it to use this syntax without the TODO as that seems to work.

Reply:
Sorry, didn't mean to include the "TODO" 🤦‍♂️